base/src/java/nonstandard/sync/condition.d

```diff
@@ -13,597 +13,10 @@
-                {
-                    ++numReady;
-                    condReady.notify();
-                }
-            }
-            while( true )
-            {
-                synchronized( synLoop )
-                {
-                    if( numDone >= numWaiters )
-                        break;
-                }
-                Thread.yield();
-            }
-            for( int j = 0; j < numWaiters; ++j )
-            {
-                semDone.notify();
-            }
-        }
-
-        group.joinAll();
-        assert( numTotal == numWaiters * numTries );
-    }
-
-
-    void testNotifyAll()
-    {
-        auto mutex      = new Mutex;
-        auto condReady  = new Condition( mutex );
-        int  numWaiters = 10;
-        int  numReady   = 0;
-        int  numDone    = 0;
-        bool alert      = false;
-
-        void waiter()
-        {
-            synchronized( mutex )
-            {
-                ++numReady;
-                while( !alert )
-                    condReady.wait();
-                ++numDone;
-            }
-        }
-
-        auto group = new ThreadGroup;
-
-        for( int i = 0; i < numWaiters; ++i )
-            group.create( &waiter );
-
-        while( true )
-        {
-            synchronized( mutex )
-            {
-                if( numReady >= numWaiters )
-                {
-                    alert = true;
-                    condReady.notifyAll();
-                    break;
-                }
-            }
-            Thread.yield();
-        }
-        group.joinAll();
-        assert( numReady == numWaiters && numDone == numWaiters );
-    }
-
-
-    void testWaitTimeout()
-    {
-        auto mutex      = new Mutex;
-        auto condReady  = new Condition( mutex );
-        bool waiting    = false;
-        bool alertedOne = true;
-        bool alertedTwo = true;
-
-        void waiter()
-        {
-            synchronized( mutex )
-            {
-                waiting    = true;
-                // we never want to miss the notification (30s)
-                alertedOne = condReady.wait( dur!"seconds"(30) );
-                // but we don't want to wait long for the timeout (10ms)
-                alertedTwo = condReady.wait( dur!"msecs"(10) );
-            }
-        }
-
-        auto thread = new Thread( &waiter );
-        thread.start();
-
-        while( true )
-        {
-            synchronized( mutex )
-            {
-                if( waiting )
-                {
-                    condReady.notify();
-                    break;
-                }
-            }
-            Thread.yield();
-        }
-        thread.join();
-        assert( waiting );
-        assert( alertedOne );
-        assert( !alertedTwo );
-    }
-
-
-    unittest
-    {
-        testNotify();
-        testNotifyAll();
-        testWaitTimeout();
-    }
-}
-
-}
+deprecated("Use core.sync.exception.SyncError instead")
+public alias SyncException = SyncError;
```

Comment:
In theory `Exception` and `Error` have different semantics. For example, an `Error` indicates a logical error/bug in the code and should not be caught. In practice, it's never been enforced and `Error` can be caught. `Exception` indicates a more of an exception in the environment which is out of control of the program, i.e. a missing file, socket connection error and so on.